xdsl/dialects/arm/ops.py

```diff
@@ -10,7 +18,43 @@ class ARMOperation(IRDLOperation, ABC):
     Base class for operations that can be a part of ARM assembly printing.
     """
 
-    ...
+    comment = opt_attr_def(StringAttr)
+    """
+    An optional comment that will be printed along with the instruction.
+    """
+
+
+@irdl_op_definition
+class DSMovOp(ARMOperation):
```

Comment:
If we're going with D for destination, and S for source, maybe it would be worth following this in the names of the results and operands? so something like `d = result_def...` `s = operand_def...`